### PR TITLE
[ICDS Dashboard][Data Pull] Data Pull: Underage Pregnancies - Feb 2020

### PR DESCRIPTION
**Pull request by ShubhamBansal1997**
##### SUMMARY
https://app.asana.com/0/1179234577945697/1180530644711913/f

### CODE REVIEW

File: custom/icds_reports/utils/data_pull.sql [hunk 1]
```diff
@@ -0,0 +1,17 @@
+SELECT
+    awc.state_name,
+    awc.district_name,
+    awc.block_name,
+    awc.supervisor_name,
+    awc.awc_name,
+    ccs.person_name,
+    CASE WHEN ccs.pregnant=1 THEN 'PW' ELSE 'LW' END,
+    age(NOW(), ccs.dob) as age
```
Comment (Rohit25negi) [inline custom/icds_reports/utils/data_pull.sql:9]: there is already a column age_in_months you can use that

File: custom/icds_reports/utils/data_pull.sql [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    FROM "ccs_record_monthly" ccs
+    LEFT JOIN "awc_location" awc ON (
+        awc.doc_id = ccs.awc_id
+        AND awc.supervisor_id = ccs.supervisor_id
+    )
+    WHERE
+    ccs.age_in_months >= 132 AND ccs.age_in_months<168
+    AND (ccs.pregnant=1 OR ccs.lactating=1);
```
Comment (Rohit25negi) [inline custom/icds_reports/utils/data_pull.sql:17]: could you exclude test states?

File: custom/icds_reports/utils/data_pull.sql [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    awc.state_name,
+    awc.district_name,
+    awc.block_name,
+    awc.supervisor_name,
+    awc.awc_name,
+    ccs.person_name,
+    CASE WHEN ccs.pregnant=1 THEN 'PW' ELSE 'LW' END,
+    age(NOW(), ccs.dob) as age
+    FROM "ccs_record_monthly" ccs
+    LEFT JOIN "awc_location" awc ON (
```
Comment (Rohit25negi) [inline custom/icds_reports/utils/data_pull.sql:11]: good to make it inner join

File: custom/icds_reports/utils/data_pull.sql [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+        awc.doc_id = ccs.awc_id
+        AND awc.supervisor_id = ccs.supervisor_id
```
Comment (Rohit25negi) [inline custom/icds_reports/utils/data_pull.sql:13]: it only needs to be on agg_level 5

File: custom/icds_reports/utils/data_pull.sql [hunk 5]
```diff
@@ -0,0 +1,17 @@
+SELECT
+    awc.state_name,
+    awc.district_name,
+    awc.block_name,
+    awc.supervisor_name,
+    awc.awc_name,
+    ccs.person_name,
+    CASE WHEN ccs.pregnant=1 THEN 'PW' ELSE 'LW' END,
```
Comment (Rohit25negi) [inline custom/icds_reports/utils/data_pull.sql:8]: you can use ccs_status column here.

**Review by Rohit25negi — CHANGES_REQUESTED** [review]
I think the query could be a bit more simpler and accurate.
could you put the cost as well

File: custom/icds_reports/utils/data_pull.sql (hunk 4 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/utils/data_pull.sql:13]: updated

File: custom/icds_reports/utils/data_pull.sql (hunk 1 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/utils/data_pull.sql]: updated

File: custom/icds_reports/utils/data_pull.sql (hunk 2 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/utils/data_pull.sql]: updated

File: custom/icds_reports/utils/data_pull.sql (hunk 3 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/utils/data_pull.sql]: updated

File: custom/icds_reports/utils/data_pull.sql (hunk 5 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/utils/data_pull.sql]: updated

**Review by Rohit25negi — APPROVED** [review]

**Pull request closed without merging**